sdks/python/apache_beam/transforms/core.py

```diff
@@ -1373,11 +1372,104 @@ def infer_output_type(self, unused_input_type):
   def expand(self, pbegin):
     assert isinstance(pbegin, pvalue.PBegin)
     self.pipeline = pbegin.pipeline
-    return pvalue.PCollection(self.pipeline)
+    ouput_type = (self.get_type_hints().simple_output_type(self.label) or
+                  self.infer_output_type(None))
+    coder = typecoders.registry.get_coder(ouput_type)
+    source = self._create_source(self.value, coder)
+    return pbegin.pipeline | Read(source).with_output_types(ouput_type)
 
   def get_windowing(self, unused_inputs):
     return Windowing(GlobalWindows())
 
+  @staticmethod
+  def _create_source(values, coder, is_serialized=False):
+    from apache_beam import io
+
+    class _CreateSource(io.iobase.BoundedSource):
+      def __init__(self, values, coder, is_serialized=False):
+        self._coder = coder
+        self._serialized_values = []
+        self._total_size = 0
+        if is_serialized:
+          self._serialized_values = values
+        else:
+          for value in values:
```

Comment:
Slightly better `self._serialized_values = map(self_coder.encode, values)`. Let me know if this is not pythonic.